robyn/dependency_injection.py

```diff
@@ -0,0 +1,49 @@
+""" This is Robyn's dependency injection file.
+"""
+from robyn.robyn import (
+    Response,
+    Request,
+)
+
+
+class DependencyMap:
+    def __init__(self):
+        #'request' and 'response' mappings are needed for when constructing deps_to_pass in router.py
+        self._dependency_map = {"ALL_ROUTES": {"request": Request, "response": Response}}
+
+    def add_route_dependency(self, route, **kwargs):
+        """ Adds a dependency to a route.
+        
+        Args:
+            route (str): The route to add the dependency to.
+            kwargs (dict): The dependencies to add to the route.
+        """
+        if route not in self._dependency_map:
+            self._dependency_map[route] = {}
+
+        self._dependency_map[route].update(**kwargs)
+
+    def add_global_dependency(self, **kwargs):
+        """ Adds a dependency to all routes.
+
+        Args:
+            kwargs (dict): The dependencies to add to all routes.
+        """
+        for element in self._dependency_map:
+            self._dependency_map[element].update(**kwargs)
+
+    def get_dependencies(self, route=None):
+        """ Gets the dependencies for a route.
+
+        Args:
+            route (str): The route to get the dependencies for.
+        """
+        if route:
+            if route in self._dependency_map:
```

Comment:
There is a logic flaw here, if route is passed but it does not exist in `self._dependency_map` this will return None.

Reply:
There is also discrepancy in the API for setting and getting dependencies, for setting there are 2 different functions, `set_route_dependency` and `set_global_dependency` but for getting there is one common function - I feel that this should be aligned, either way is fine IMO, but should be the same.